internal/providers/microsoft_entra_id/microsoft-entra-id.go

```diff
@@ -89,6 +89,32 @@ func (p MSEntraIDProvider) GetGroups(token *oauth2.Token) ([]group.Info, error)
 	return groups, nil
 }
 
+// AuthenticationModes returns the generic authentication modes supported by the provider.
+func (p MSEntraIDProvider) AuthenticationModes(sessionMode, _ string, tokenExists bool, currentAuthStep int) ([]string, error) {
```

Comment:
please add a comment that we didn’t take into account password expiration here and token validity which will impact the list of returned auth modes.